src/yetibot/core/db/util.clj

```diff
@@ -138,44 +138,66 @@
      {}
      qs)))
 
-(defn query
-  "Query with WHERE"
-  [table {;; provide either where/map
+(defn generate-sql-query
```

Comment:
Huge +1 – I had hoped to do this at some point, so thank you!

The result of this function is pure data - testable, manipulatable, repl friendly, human friendly etc. Definitely makes sense to get at it independent from the side effects it's intended to produce.